ROADMAP.md

```diff
@@ -0,0 +1,155 @@
+
+The [Appendix](#-appendix) section describes a project's high-level motivation, goals, and lists sub-projects.
+
+Each Appendix header is linked to a GitHub Epic. Latest information on progress can be found in the Epics and child issues.
+
+### Done criteria
+The "Definition of Done" for projects/sub-projects that involve writing new protocols/ modify existing ones usually consist of the following:
+- If a specification change is required:
+    - [ ] Spec is merged and classified as "Candidate Recommendation"
+    - [ ] (by virtue of the above) At least one major reference implementation exists
+- [ ] A well established testing criteria is met (defined at the outset of the project including but not limited to testing via Testground, compatibility tests with other implementations in the Release process, etc.)
+- [ ] Public documentation (on docs.libp2p.io) exists
+
+Supporting projects (such as testing or benchmarking) may have different criteria.
+
+### Benchmarking and Testing
+As mentioned in our [vision](#vision), performance and stability are core libp2p tenets. Rigorous benchmarking and testing help us uphold them. Related projects are listed in the [libp2p/test-plans roadmap](https://github.com/libp2p/test-plans/pull/44) and the [testground/testground roadmap](https://github.com/testground/testground/pull/1484). Our major priorities in Q4’22 and Q1’23 are:
+- [interoperability testing](https://github.com/libp2p/test-plans/issues/53) (across implementations & versions and between transports, muxers, & security protocols)
+- [add a browser environment test harness to support testing browser features](https://github.com/testground/testground/issues/1386)
+- test DHT Server Mode at scale (testbed of at least >20 nodes; ideally 100/1000+) in Testground
+- performance benchmark js-libp2p using Testground (create a benchmark suite to run in CI)
+
+These projects are parallel workstreams, weighed equally with roadmap items in this document. Some efforts like interoperability testing have a higher priority than implementation projects. The js-libp2p co-owns these efforts with the go-libp2p, rust-libp2p, and Testground/IPDX teams.
+
+## 🛣️ Milestones
+### 2022
+
+#### Early Q4 (October)
+- [A.1 📺 WebTransport](#1-webtransport)
+  
+#### Mid Q4 (November)
+- [C.1 🔮 Benchmark transfer performance in CI](#1-benchmark-transfer-performance-in-ci)
+
+#### End of Q4 (December)
+- [A.2 📺 WebRTC: Browser to Server](#2-webrtc-browser-to-server)
+- [B.1 🥊 AutoNat](#1-autonat)
+- [B.2 🥊 Circuit Relay v2](#2-circuit-relay-v2)
+
+### 2023
+
+#### Mid Q1 (February)
+- [C.2 🔮 Add introspection to js-libp2p](#2-add-introspection-to-js-libp2p)
+
+#### End of Q1 (March)
+- [B.3 🥊 Add QUIC Transport](#3-add-quic-transport)
+
+#### Early Q2 (April)
+- [B.4 🥊 Hole Punching](#4-hole-punching)
+	- 🎉 Estimated Project Completion
+
+### Up Next
+- [A.3 📺 WebRTC: Browser to Browser](#3-webrtc-browser-to-browser)
```

Comment:
I would vote for WebRTC browser to browser ahead of QUIC transport because:
1. It enables net-new capabilities that can't be accomplished via other means
2. Reduces maintenance burden by getting rid of webrtc-star and webrtc-direct. 

My guess (I don't have user anecdotes) is that users will find more value being able to build broser-to-browser apps than nodejs apps having hole punching.

Reply:
We can build it with the WebRTC primitives available today as well so less risky from a project standpoint, no need to wait for QUIC support or try to hack it together ourselves.